internal/apis/certmanager/types_issuer.go

```diff
@@ -284,6 +284,14 @@ type ServiceAccountRef struct {
 	// consisting of the issuer's namespace and name is always included.
 	// +optional
 	TokenAudiences []string
+
+	// EnableKubeAudience will include the default Kubernetes audience as the
+	// service account's token audience. This is useful when the short lived service
+	// account token is used to authenticate with Vault and Vault does not have JWT validation token configured.
+	// This enables Vault to use the service account token as a JWT token with the audience set to the Kubernetes API server.
+	// If unset the audience defaults to `false`
```

Comment:
"Default audience" slightly confuses me. I think we should rather say that it will default to kube-apiserver's issuer URL that is defined by kube-apiserver's `--service-account-issuer` and that is visible by running:

```bash
kubectl get --raw /.well-known/openid-configuration | jq .issuer -r
```

What did you mean by "If unset the audience defaults to `false`"?

I'm unsure about what you mean by "and Vault does not have JWT validation token configured". Do you mean that "and you are unable to use Vault's JWT/OIDC auth"?

I suggest something along the lines of:

```suggestion
	// EnableKubeAudience, when set, includes kube-apiserver's service account
	// issuer URL to the service account's token audience. This way, the
	// short-lived service account tokens created by cert-manager and used to
	// authenticate with Vault can also be used by Vault to authenticate with
	// kube-apiserver when Vault does its TokenReview check. This is useful when
	// using Vault's Kubernetes auth with an out-of-cluster Vault and when you
	// cannot use Vault's JWT/OIDC auth instead.
```

Reply:
What I have found is JWT issuer does not equals to the k8s API audience in many cases, e.g. in AKS issuer would be something like `https://westeurope.oic.prod-aks.azure.com/<GUID>/<GUID>` where the aud would be the AKS API endpoint.

> What did you mean by "If unset the audience defaults to false"?

if not set then it would be the default value of the flag, I guess this can be dropped as this is a default Go behaviour either way.

> I'm unsure about what you mean by "and Vault does not have JWT validation token configured".

This is a scenario where we configure a long lived k8s SA JWT token in the Vault k8s auth backend so it is used in the call to the k8s API service to validate if a client token is still valid. In the short lived token scenario client token is used in the call to the k8s API service.

Will this look better?

```txt
EnableKubeAudience, when set, includes the audience expected by
the kube-apiserver's controller to the service account's token audience. This way, the
short-lived service account tokens created by cert-manager and used to
authenticate with Vault can also be used by Vault to authenticate with
kube-apiserver when Vault does its TokenReview check. This is useful when
using Vault's Kubernetes auth with an out-of-cluster Vault and when you
cannot use Vault's JWT/OIDC auth instead.
```